core/mock_test.go

```diff
@@ -181,6 +181,23 @@ func (m mockBackend) BuildRoundChangeMessage(
 	}
 }
 
+func (m mockBackend) HasQuorum(view *proto.View, messages []*proto.Message) bool {
+	quorum := m.Quorum(view.Height)
+
+	if len(messages) > 0 {
+		switch messages[0].GetType() {
+		case proto.MessageType_PREPREPARE:
+			return len(messages) <= 1
+		case proto.MessageType_PREPARE:
+			return len(messages) < int(quorum)-1
+		case proto.MessageType_ROUND_CHANGE, proto.MessageType_COMMIT:
+			return len(messages) < int(quorum)
```

Comment:
Strange that function is named like this but instead is checking whether we haven't reached quorum.

Therefore suggesting changing logical conditions as following:

```suggestion
		switch messages[0].GetType() {
		case proto.MessageType_PREPREPARE:
			return len(messages) > 1
		case proto.MessageType_PREPARE:
			return len(messages) >= int(quorum)-1
		case proto.MessageType_ROUND_CHANGE, proto.MessageType_COMMIT:
			return len(messages) >= int(quorum)
```

Btw, do we need all these cases here? Seems like we need only for `PREPARE` and `ROUND_CHANGE` messages if I'm correct (was looking where we are checking `HasQuorum` within the `if` statements)

Reply:
You're right conditions are reversed, thx! 

The other extra cases are for subscriptions they also depend on those extra message types.